src/components/MultiHopTrade/components/FiatRamps/FiatInput.tsx

```diff
@@ -71,13 +100,23 @@ export const FiatInput: React.FC<FiatInputProps> = ({
     number: { localeParts },
   } = useLocaleFormatter()
 
+  const formattedValueLengthRef = useRef<number>(0)
+  const [formattedValueLength, setFormattedValueLength] = useState(0)
+
   const fiatSymbol = useMemo(() => {
     if (!showPrefix) return ''
     return selectedFiatCurrency.symbol
   }, [selectedFiatCurrency, showPrefix])
 
   const handleAmountChange = useCallback(
     (values: NumberFormatValues) => {
+      // Update the formatted value length
```

Comment:
q: does this also work on scenarios other than manual input? e.g quick amounts etc